### PR TITLE
add TLS support for metrics collector and scaling engine endpoint

### PR DESCRIPTION
**Pull request by boyang9527**
[#121340317]

- add SSL config for metrics collector and scaling engine
- add SSL support for http servers
- test https in main

### CODE REVIEW

File: src/autoscaler/metricscollector/cmd/metricscollector/metricscollector_suite_test.go [hunk 1]
```diff
@@ -227,3 +265,75 @@ func marshalMessage(message *events.Envelope) []byte {
 
 	return data
 }
+
+var testCaCert = []byte(`
```
Comment (fraenkel) [inline src/autoscaler/metricscollector/cmd/metricscollector/metricscollector_suite_test.go]: Why aren't these just test assests as files?

File: src/autoscaler/metricscollector/config/config.go [hunk 2]
```diff
@@ -23,11 +24,13 @@ var defaultCfConfig = cf.CfConfig{
 }
 
 type ServerConfig struct {
-	Port int `yaml:"port"`
+	Port      int  `yaml:"port"`
+	EnableSSL bool `yaml:"enable_ssl"`
```
Comment (fraenkel) [inline src/autoscaler/metricscollector/config/config.go]: Why isn't the SSL parts just here as an option? If present, we do SSL, if not, we don't.

Comment (boyang9527) [inline src/autoscaler/metricscollector/config/config.go]: removed EnableSSL setting

File: src/autoscaler/metricscollector/cmd/metricscollector/metricscollector_suite_test.go (hunk 1 above)
Comment (boyang9527) [inline src/autoscaler/metricscollector/cmd/metricscollector/metricscollector_suite_test.go]: moved the certificates/key  to files

**Comment by boyang9527** [comment]
@fraenkel  changes submitted, pls review.

**Comment by fraenkel** [comment]
Merged

**Pull request closed without merging**